prog/pwm/fancontrol

```diff
@@ -390,6 +393,39 @@ function pwmdisable()
 		return 0
 	fi
 
+	# Try to restore pwmN and pwmN_enable value to the same state as before fancontrol start
+	# restoring the pwmN value is tried first, before the pwmN_enable mode switch. Some chips seem to need this to properly restore fan operation, when activating automatic (2) mode.
+	if [ -v PWMORIGVALUES[${1}] ]
+	then
+		local PWM_ENABLE_ORIG_STATE=`echo ${PWMORIGVALUES[${1}]} | cut -d ';' -f1`
+		local PWM_ORIG_STATE=`echo ${PWMORIGVALUES[${1}]} | cut -d ';' -f2`
+		#restore the original pwmN value
+		if [ "$DEBUG" != "" ]
+		then
+			echo "Restoring ${1} original value of $PWM_ORIG_STATE"
+		fi
+		echo $PWM_ORIG_STATE > ${1} 2> /dev/null
+        #restore the pwmN_enable value, if it is not 1. 1 is already set through fancontrol and setting it again might just reset the pwmN value.
+        if [ $PWM_ENABLE_ORIG_STATE -ne 1 ]
+        then
+            if [ "$DEBUG" != "" ]
+            then
+                echo "Restoring $ENABLE original value of $PWM_ENABLE_ORIG_STATE"
+            fi
+            echo $PWM_ENABLE_ORIG_STATE > $ENABLE 2> /dev/null
+        fi
+        #if pwmN_enable is not manual, do not check value of pwmN, as it might already have been altered by the chip.
+        if [ `cat $ENABLE` -eq $PWM_ENABLE_ORIG_STATE ] && [ $PWM_ENABLE_ORIG_STATE -ne 1 ]
+        then
+            return 0
+        fi
+        #if pwmN_enable is manual, check if setting the original pwm value actually worked
+        if [ $PWM_ENABLE_ORIG_STATE -eq 1 ] && [ `cat $ENABLE` -eq 1 ] && [ `cat ${1}` -eq $PWM_ORIG_STATE ]
```

Comment:
The original state could also have been 0 (disabled). Overall I am not sure if this check makes sense. Since we no longer have fan control, it might be better to revert to the original code here (enable=1 might not be sufficient even if the original value was restored).